### ISSUE TITLE
Reviewers can comment on submissions during peer review

### ISSUE THREAD

**Issue by elisemarika** [MetricsDAO/xyz#604]
**Job to be Done**
As a Reviewer in the Metrics app, I want to be able to leave comments on other analysts' submissions when scoring them, so I can provide more feedback/suggestions/reasoning relating to the score I give them during peer review, and help them learn/grow as analysts and level up future submissions.

**Requirements**
- Must have: Users can optionally leave comments on submissions they score << Reviewers only, or anyone?
- Confirm: Users must score a submission to be able to comment << should we allow comments without scoring? imo no
- Confirm: Users don't need to signal to comment
- Nice to have: Users (Reviewers only, or anyone?) can reply to comments with nested commenting. imo makes sense that an Analyst should be able to review to a comment they received from someone to have a back and forth dialogue

**Design**
- Should commenting happen in the same step as submitting a score? Or as a separate, optional step 2? Or as a totally separate step

**Comment by masonchain** [comment]
I think this makes sense to have and all your requirements outline it well. None of this has to be on chain or actually impact the protocol or chain at all. Reviews and comments could be stored on IPFS and their hashes in the app db.

If we did want to include writing the IPFS hash of a review when a reviewer does grade a submission, that is entirely possible. Any other comment should definitely not be written to the chain though. 

@nftchance What do you think? All IPFS?

**Comment by thekakamora** [comment]
"Users can optionally leave comments on submissions they score" - in our existing system we made it as mandatory as several reviewers were marking posts as spam without giving any explanation. Also few reviewers  mark a submission as excellent or very bad without any context. It will be helpful if we have option to mark comments as mandatory. Preferably configurable while launching the challenges. 


Also please consider ability to upload proofs like screenshots so that reviewers are empowered to backup their claims on plagiarized posts, empty dashboards, etc.

**Comment by ltirrell** [comment]
I agree with kakamora that it should probably be mandatory to include comments. Ideally, submitting a comment should be in the same step as submitting a score during the review process (the less interactions to get reviews done the better)

**Comment by elisemarika** [comment]
@thekakamora @ltirrell super helpful, thank you. please share feedback on this updated set too
- Users must leave comments when scoring submissions
- Users can _only_ leave comments when scoring submissions << or should they also be able to in isolation
- Analysts can only reply to comments left on their own submission << or should they also be able to reply to other comments

**Comment by thekakamora** [comment]
If leaving comments outside of reviews is an onchain txns, I'm not sure how many will be willing to comment. Nonetheless I think it is helpful to keep the comments / replies open for everyone so that community can collaborate & debate when necessary.
 
Users can only leave comments when scoring submissions << or should they also be able to in isolation - _I vote for ability to leave comments on isolation._ 

Analysts can only reply to comments left on their own submission << or should they also be able to reply to other comments . _I vote for "be able to reply to other comments"_

**Comment by drakedanner** [comment]
I agree with @thekakamora and @ltirrell generally.

- Reviewers _should_ comment when Reviewing 
- Comments in isolation are a nice to have
- Analyst replies could be to any comment

With this understanding, should comments simply be open to everyone? OR do we think it should be limited to participants? If so why?

Comments should exist at the app layer, not be onchain, and therefore not create friction in participation.

**Comment by ewharton** [comment]
Agree, reviewers should be required to comment ("rationale"). Unexplained bad reviews aren't constructive and can quickly breed ill will to the whole system.

Anyone should able to comment and reply. Discussion is how a community learns together.

**Comment by elisemarika** [comment]
Elise/Drake 4/11 (let's not build a forum)

- Reviewers can optionally comment when Reviewing
- Anyone can optionally comment when viewing Submissions (not gated)
- Anyone can optionally reply to someone elses comment (threaded) << only comments that also had a score?
- Comments associated with scores should be easily visible/connected

**Comment by Sunslinger** [comment]
<img width="1047" alt="Screenshot 2023-04-28 at 12 04 31 PM" src="https://user-images.githubusercontent.com/80657355/235209649-fccadcb6-9131-47e6-8509-c730ed0f1a53.png">

Testing - providing community feedback to illuminate an issue

**Comment by Sunslinger** [comment]
It works! I'll start providing examples of community feedback that highlight the importance/prioritization of specific issues.

**Comment by drakedanner** [comment]
fyi @sebruiz @thehannahlerman moving this one into ready for dev :)

This will utilize the URI on Reviews in V2!
